recipes/perl/rbac_roles/rbac_role_templates.pl

```diff
@@ -0,0 +1,793 @@
+        'description' => 'Provides all the permissions necessary to manage Resiliency Platform for VMware assets.',
+        'template'    => \%vrp_role_operation_map,
+    },
+    'RHV Administrator'        => {
+        'name'        => 'RHV Administrator',
+        'description' => 'Provides all the permissions necessary to manage protection for RedHat Virtualization VMs through Protection Plans.',
+        'template'    => \%rhv_role_operation_map,
+    },
+    'Cloud Administrator' => {
+        'name'        => 'Cloud Administrator',
+        'description' => 'Provides all the permissions necessary to manage protection of cloud assets using Protection Plans.',
+        'template'    => \%cloud_role_operation_map,
+    },
+    'MS-SQL Administrator'     => {
+        'name'        => 'MS-SQL Administrator',
+        'description' => 'Provides all the permissions necessary to manage protection for Microsoft SQL Server databases using Protection Plans.',
+        'template'    => \%mssql_role_operation_map,
+    },
+    'Storage Administrator'    => {
+        'name'        => 'Storage Administrator',
+        'description' => 'Provides all the permissions necessary to configure and manage disk-based storage and cloud storage for NetBackup.',
+        'template'    => \%storage_role_operation_map,
+    },
+    'Security Administrator'   => {
+        'name'        => 'Security Administrator',
+        'description' => 'Security administrator with privileges for performing actions including access control, certificates, security settings etc.',
+        'template'    => \%security_role_operation_map,
+}
+);
+
+# Parse command line options
+my ($help, $man, $debug);
+GetOptions(
+    'help|h!'          => \$help,
+    'man|m!'           => \$man,
+    'debug|d!',        => \$debug,
+    'list_templates'   => \$list_templates,
+    'hostname|host=s'  => \$hostname,
+    'username|user=s'  => \$username,
+    'password|pass=s'  => \$password,
+    'domain_type=s'    => \$domain_type,
+    'domain_name=s'    => \$domain_name,
+    'token=s'          => \$token,
+    'create_templates' => \$create_all,
+    'template=s'       => \$template,
+    'name=s'           => \$name,
+    'description=s'    => \$description
+) or pod2usage(2);
+
+if ($help) {
+    pod2usage(1);
+    exit;
+}
+
+if ($man) {
+    pod2usage(-verbose => 2);
+    exit;
+}
+
+if ($list_templates) {
+    print_templates();
+    exit;
+}
+
+unless ($create_all or $template) {
+    pod2usage(1);
+    exit 1;
+}
+
+if ($create_all and $template) {
+    print "Cannot specify both --create_templates and --template for role creation.\n";
+    pod2usage(1);
+    exit 1;
+}
+
+if ($create_all and ($name or $description)) {
+    print "--name and --description are only valid with the --template option.\n";
+    pod2usage(1);
+    exit 1;
+}
+
+if ($template) {
+    my $found = 0;
+    foreach my $role (keys %role_template_data) {
+        if ($template =~ /$role/) {
```

Comment:
Here, `$template` is compared to the hash key with a regular expression, so it will match whenever there's a key that's a substring of the template. Once it's found, though, line 508 assumes the template was an exact match. If an exact match is required, then that's what we should use to search.

```suggestion
        if ($template eq $role) {
```

But then we don't even need the loop. We can just check directly:

```perl
my $found = defined $role_template_data{$template};
```